cmake/HPX_GeneratePackage.cmake

```diff
@@ -60,7 +60,7 @@ if(HPX_WITH_STATIC_LINKING)
   set(HPX_PKG_LIBRARIES "\${libdir}/libhpx.a ${HPX_PKG_LIBRARIES}")
   set(HPX_PKG_DEBUG_LIBRARIES "\${libdir}/libhpxd.a ${HPX_PKG_DEBUG_LIBRARIES}")
 else()
-  set(HPX_CONF_LIBRARIES "general;hpx;general;hpx_init;${HPX_LIBRARIES}")
+  set(HPX_CONF_LIBRARIES "general;hpx_init;hpx;general;${HPX_LIBRARIES}")
```

Comment:
This will break cmake dependency handling. the 'general;' has to prefix the library name. If you need to change the linking order, please modify this to 
```
set(HPX_CONF_LIBRARIES "general;hpx_init;general;hpx;${HPX_LIBRARIES}") 
```

Reply:
Ok, thank you for pointing that out. I made the changes as suggested.